geoh5py/objects/object_base.py

```diff
@@ -224,8 +226,30 @@ def clip_by_extent(
             with shape(2, 3) defining the top and bottom limits.
         :param attributes: Dictionary of attributes to clip by extent.
         """
+        if not any(mask_by_extent(bounds, self.extent)) and not any(
+            mask_by_extent(self.extent, bounds)
+        ):
+            return None
+
         new_entity = self.copy(parent=parent, copy_children=copy_children)
-        return new_entity
+        return new_entity.clip_by_extent(bounds)
+
+    def clip_by_extent(self, bounds: np.ndarray) -> ObjectBase | None:
+        """
+        Find indices of cells within a rectangular bounds.
+
+        :param bounds: shape(2, 2) Bounding box defined by the South-West and
+            North-East coordinates. Extents can also be provided as 3D coordinates
+            with shape(2, 3) defining the top and bottom limits.
+        :param attributes: Dictionary of attributes to clip by extent.
+        """
+
+        # TODO Clip entity within bounds.
```

Comment:
**chore:** is there a JIRA issue to track this?

Reply:
Riding with the devil!!! GEOPY-666